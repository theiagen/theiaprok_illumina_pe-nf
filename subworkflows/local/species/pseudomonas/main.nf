include { PASTY } from '../../../../modules/local/species_typing/pseudomonas/pasty/main'

workflow PSEUDOMONAS_AERUGINOSA_SPECIES_TYPING {

    take:
    ch_pseudomonas // Channel of tuples [meta, assembly, reads, species]

    main:

    ch_versions = Channel.empty()
    ch_pasty_results = Channel.empty()
    ch_value_results = Channel.empty()

    ch_assembly = ch_pseudomonas.map { meta, assembly, reads, species -> [meta, assembly] }

    PASTY (
        ch_assembly
    )
    ch_value_results = ch_value_results.mix(PASTY.out.pasty_value_results)
    ch_pasty_results = PASTY.out.pasty_summary_tsv
    ch_versions = ch_versions.mix(PASTY.out.versions)

    emit:
    pasty_results = ch_pasty_results
    value_results = ch_value_results
    versions = ch_versions
}