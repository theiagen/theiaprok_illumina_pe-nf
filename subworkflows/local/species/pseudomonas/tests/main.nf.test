nextflow_workflow {
    name "Test Subworkflow PSEUDOMONAS_AERUGINOSA_SPECIES"
    script "../main.nf"
    workflow "PSEUDOMONAS_AERUGINOSA_SPECIES_TYPING"

    test("pseudomonas_aeruginosa - pasty characterization") {
        setup {
            run("GUNZIP") {
                script "../../../../../modules/nf-core/gunzip/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [id:'test_pseudomonas', single_end:false],
                        file('https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/765/GCF_000006765.1_ASM676v1/GCF_000006765.1_ASM676v1_genomic.fna.gz', checkIfExists: true)
                    ])
                    """
                }
            }
        }
        
        when {
            workflow {
                """
                input[0] = GUNZIP.out.gunzip
                    .map { meta, assembly -> 
                        [
                            meta, 
                            assembly, 
                            [],
                            "Pseudomonas aeruginosa"
                        ]
                    }
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.pasty_results,
                    workflow.out.versions
                ).match() }
            )
        }
    }
}