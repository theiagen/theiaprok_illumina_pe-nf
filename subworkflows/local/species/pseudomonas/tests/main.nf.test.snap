{
    "pseudomonas_aeruginosa - pasty characterization": {
        "content": [
            [
                [
                    {
                        "id": "test_pseudomonas",
                        "single_end": false
                    },
                    [
                        "test_pseudomonas.blastn.tsv:md5,ec58783a0ded5271804a072d74a4a0d0",
                        "test_pseudomonas.details.tsv:md5,e729ca7da5a77b77e8e1cd1727061728",
                        "test_pseudomonas.tsv:md5,4e394a62e98d4cc5cfb77fac7e44b42d"
                    ]
                ]
            ],
            [
                "versions.yml:md5,d3377b3a3a370915f1c61a903378c013"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.5"
        },
        "timestamp": "2025-07-02T20:35:58.246034"
    }
}